src/dataset.js

```diff
@@ -240,11 +265,18 @@ Dataset.prototype.createQueryJob = function(options, callback) {
     };
   }
 
-  options = extend(true, {}, options, {
-    defaultDataset: {
-      datasetId: this.id,
+  options = extend(
+    true,
+    {
+      location: this.location,
```

Comment:
Can we avoid having this option, and just let the user either provide the location from the `bigquery.dataset()` constructor, or have it set already for the objects we give to the user?

From the req sheet:

> Allow the user to optionally specify a location for job operations.  Consider having one flag that can be used for dataset creation (the existing location field) and the job operations.

This same question applies to the other objects and methods as well.

Reply:
I've reviewed some of the other languages implementations and there is similar functionality like this in the other clients. That being said, I think we should keep this.

Reply:
cc @tswast @jjolma 

For Node, if a property needs to be changed, we recommend the user creates a new instance, e.g.

```js
const dataset1 = bigquery.dataset('name', { location: 'location-1' })
const dataset2 = bigquery.dataset('name', { location: 'location-2' })
```

Then, all operations on that object are correctly scoped:

```js
dataset1.createQueryJob(...) // uses `location-1`
dataset2.createQueryJob(...) // uses `location-2`
```

This is similar to how the user can use different project IDs, auth credentials, etc:

```js
const BigQuery = require('@google-cloud/bigquery')
const bigQueryProject1 = new BigQuery({ project: 'project-id-1' })
const bigQueryProject2 = new BigQuery({ project: 'project-id-2' })
```

Reply:
Correct, I think we should only support passing into the constructor. It maintains the correct hierarchy, as the pairing of a `location` to a Dataset is similar to the pairing of a `name`, in that they both need to be known up front. Lazy-setting of the location feels similar to lazy-setting a name. In other words, we don't allow the following pattern, so I don't see why we should allow it for location:

```js
const dataset = bigquery.dataset();
const [metadata] = await dataset.getMetadata({ name: 'dataset-name' });
```

Reply:
I'm not sure I agree with that stance, `location` being a query parameter, as opposed to a path parameter, makes me think it's ok to configure it in this fashion. I'm not opposed to removing it, but I'd prefer to hear from the BigQuery team before doing so, just in case they think there is some value in it.

Reply:
Yes, location is something that needs to be set before the API call to create the dataset. In an ideal world it would have been a path parameter, but such a change wouldn't be backwards compatible.

Note: the [table encryption key](https://cloud.google.com/bigquery/docs/customer-managed-encryption) is similar to this in that it can't be set if you create a table without one (though unlike location, it can be changed to a new value after the fact)

Reply:
Very well! I've went ahead and removed this. @stephenplusplus PTAL